.cirrus.yml

```diff
@@ -1,26 +1,153 @@
 ---
 
-test_task:
-  alias: "build_and_test"
-  name: "Build and Test"
-  container:
-    image: quay.io/libpod/netavark-devel
-  install_script: make all
-  test_script: make test
+# Format Ref: https://cirrus-ci.org/guide/writing-tasks/
+
+# Main collection of env. vars to set for all tasks and scripts.
+env:
+    # Actual|intended branch for this run
+    DEST_BRANCH: "main"
+    # The default is 'sh' if unspecified
+    CIRRUS_SHELL: "/bin/bash"
+    # Location where source repo. will be cloned
+    CIRRUS_WORKING_DIR: "/var/tmp/netavark"
+    # Rust package cache also lives here
+    CARGO_HOME: "/usr/local/cargo"
```

Comment:
Are you sure you are caching the right directory?

It uses $HOME/.cargo AFAIK, so /root/.cargo if you run as root.

Reply:
According my my reading of the cargo (command) docs, setting `$CARGO_HOME` should override this.   The update I just made should allow us to confirm this env. var. is really being set at built-time vs only setup-time.

Reply:
Update: Okay, it's certainly being [set at build-time (log)](https://cirrus-ci.com/task/6629489451991040?logs=main#L7).  Hmm.

Reply:
I may need to just go in an check this out manually, since this PR adds a working `hack/get_ci_vm.sh`.

Reply:
Oh, speaking of which.  Let me make sure you've got access to use that.

Reply:
You are right `CARGO_HOME` env should work but it should compile fast if the cache is already there. If I look at the log I can that it downloads the packages which should not happen with an working cache.

Reply:
I think I understand now. The packages are cached there but this is just for the downloads. The compiled package blobs are stored in the project directory under `bin/release`.

Reply:
@Luap99 you've now got access to use the hack script.  So it appears like the `$CARGO_HOME` cache is working from one task to the next.  The latest build ran in ~7m and [validate took only ~2m (log)](https://cirrus-ci.com/task/4588795870838784?logs=main#L110).  So it's only the PR-run to PR-run cache that's flapping for some reason.  I think I'll leave it as-is for now, and dig deeper next week (after US Holiday break).

Reply:
Just checked a VM running the build task.  Indeed there's no `$HOME/.cargo` only `/usr/local/cargo`.  So this must be a Cirrus-CI Caching problem - it's not storing the content that I expect, or something is causing flapping.  In any case, I'm leaving this debugging up to the future.